### PR TITLE
Use rsdk templates and bug fix for pub_cache location issues

### LINKED ISSUE #15: Infer pub cache location for systems with unset $FLUTTER_ROOT
**Is your feature request related to a problem? Please describe.**

- Some systems use the $FLUTTER_ROOT/.pub-cache location even if the FLUTTER_ROOT variable is empty

**Describe the solution you'd like**

- Infer the pub cache location based on the flutter executable PATH

**Describe alternatives you've considered**

Rather than pulling at_app templates from the pub cache, pull them from a host.

- This solution does not work in offline
- Should be more reliable for those with internet

**Additional context**

### LINKED ISSUE #17: Move templates to rsdk
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
at_client is version locked to 2.x.x, needs to be updated to 3.x.x.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by xavierchanth**
This is 1/2 PRs for at_app, the other being [here](https://github.com/atsign-foundation/at_widgets/pull/204)

closes #15 
closes #17 

**- What I did**

- Used the .dart_tool/package_config.json file to get the location of at_app_flutter on the user's machine
- Updated the [at_app_flutter version](https://github.com/atsign-foundation/at_widgets/pull/204) to use the new rsdk release

**- How I did it**

- Local changes

**- How to verify it**
You may test using release candidate 1.1.0-rc+3, the only difference is version numbers and changelogs.
```
flutter pub global activate at_app 1.1.0-rc+3
```

**- Description for the changelog**
Use rsdk templates and bug fix for pub_cache location issues

### CODE REVIEW

**Comment by xavierchanth** [comment]
@cconstab waiting for your approval, before I merge.

File: lib/src/util/cache_package.dart
Comment (yahu1031) [inline lib/src/util/cache_package.dart:6]: Can I just know why this  .dart_tools cache is needed? Cause during version management of the app/code they will be ignored in git. Soo just curious to know why!!!

**Review by yahu1031 — APPROVED** [review]

File: lib/src/util/cache_package.dart
Comment (xavierchanth) [reply lib/src/util/cache_package.dart:6]: That specific file contains info as to where all the dependencies are locally stored on the user's machine.
So by parsing the .dart_tools/package_config.json, I can find where at_app_flutter is stored.
You don't want to commit it because it is different for every developer.

**Review by cconstab — APPROVED** [review]

**Pull request merged**
